phpstan.neon

```diff
@@ -14,5 +14,17 @@ parameters:
         - '/^Class [a-zA-Z\\]+ extends @final class PhpCsFixer\\(ConfigurationException\\InvalidConfigurationException|ConfigurationException\\InvalidFixerConfigurationException|Tokenizer\\Tokens)\.$/'
         - '/^Unsafe call to private method [a-zA-Z\\]+::[a-zA-Z]+\(\) through static::\.$/'
         - '/^\$this\(PhpCsFixer\\Tokenizer\\Tokens\) does not accept PhpCsFixer\\Tokenizer\\Token\|null\.$/'
+
+        # ignore PHPUnit data providers return type as they are not checked against the test methods anyway
+        - '#^Method PhpCsFixer\\Tests\\.+::provide.+Cases\(\) return type has no value type specified in iterable type array\.$#'
+        - '#^Method PhpCsFixer\\Tests\\.+::provide.+Cases\(\) return type has no value type specified in iterable type iterable\.$#'
+
+        # baseline
+        - '#^Method .+ has parameter .+ with no value type specified in iterable type array\.$#'
+        - '#^Method .+ has parameter .+ with no value type specified in iterable type iterable\.$#'
+        - '#^Method .+ return type has no value type specified in iterable type array\.$#'
+        - '#^Method .+ return type has no value type specified in iterable type iterable\.$#'
+        - '#^Property .+ has no type specified\.$#'
+        - '#^Property .+ type has no value type specified in iterable type array\.$#'
```

Comment:
Please consider generating `phpstan-baseline.neon` instead (or adding `count`). It may be huge, but it will ensure that no new errors of these types will be introduced.

If you decide to stick with the current approach, you can use `checkMissingIterableValueType: false` parameter.

Reply:
Letting PHPStan generate the baseline was my first approach when I previously tried to increase PHPStan's level. I think baseline feature is great but it tends to create a burden on contributors that often have to update it, so not very suitable for us here.

I added counts, thanks for the suggestion.

Reply:
I believe that baseline would be easier to deal with for contributors, as they get a suitable error message asking them to add type or remove something from baseline. With count, I think they will get something like “expected to get this error x times, but it was reported y times”, with no way to know where the change occurred.

Reply:
I propose to see how it goes and if it still makes contributions harder, we'll try something else.